src/main/java/org/opentestsystem/ap/imrt/common/repository/BaseItemRepository.java

```diff
@@ -0,0 +1,8 @@
+package org.opentestsystem.ap.imrt.common.repository;
+
+import org.opentestsystem.ap.imrt.common.model.BaseItem;
+import org.springframework.data.repository.CrudRepository;
+
+public interface BaseItemRepository extends CrudRepository<BaseItem, Integer> {
```

Comment:
The CrudRepository probably isn't appropriate here and I'll need to do something like [this](https://stackoverflow.com/questions/11107136/creating-a-read-only-repository-with-springdata?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa) before I merge

Reply:
A quick test of this polymorphic repo might be a method name like findByStimLinkStimLinkKeyItemStimKey(int stimKey);
Which should hopefully be parsed into "x.stimLink.stimLinkKey.itemStim.key"  And would confirm you could query by a sub-class' properties from the abstract repo.
Side-note after digging through the concept of a stimLink/stimLinkKey I'm a little confused.  Why would an ImrtItem just have a "stimulus" property that contains a (nullable) Stimulus?

Reply:
@wtritch 's side note is interesting.  I think we've expressed that in the database well enough, but I think we might have a simpler and more expressive object model if `ImrtItem` had an `Optional<Stimulus>` property.